katas/main/src/pokerhands/adriel/HandEvaluator.cs

```diff
@@ -0,0 +1,58 @@
+namespace PokerHands.HandsEvaluator.Adriel;
+
+using Enums.Adriel;
+using Cards.Adriel;
+using Hands.Adriel;
+using HandsRankingEvaluator.Adriel;
+using HandsKickersEvaluator.Adriel;
+using HandsMainEvaluator.Adriel;
+
+public class HandEvaluator
+{
+    public Hand Hand { get; }
+    public Ranking Ranking { get; }
+    public Card[] RankingCards { get; }
+    public Card[] Kickers { get; }
+
+    public HandEvaluator(Hand hand)
+    {
+        string tempCardsString = string.Join(" ", hand.Cards.OrderBy(c => c.Value));
+
+        Hand = new Hand(tempCardsString);
+        Ranking = HandRankingEvaluator.Evaluate(Hand);
+
+        if (IsAceLowStraightOrAceLowStraightFlush())
+        {
+            Hand.Cards.Insert(0, Hand.Cards[4]);
+            Hand.Cards.RemoveAt(5);
+        }
+
+        RankingCards = GetRankingCards();
+        Kickers = GetKickers();
+    }
+
+    private bool IsAceLowStraightOrAceLowStraightFlush()
+    {
+        return (Ranking == Ranking.Straight || Ranking == Ranking.StraightFlush) &&
+        (Hand.Contains(Value.Ace) && Hand.Contains(Value.Two) && Hand.Contains(Value.Three) &&
+        Hand.Contains(Value.Four) && Hand.Contains(Value.Five));
+    }
+
+    private Card[] GetRankingCards()
```

Comment:
These methods are good. They can be improved by using objects and inheritance to make code more extensible. This means that we will extend classes instead of modifying existing method in case we want to add new features.

We will talk more about this during classes, but so far, this is good

Reply:
I actually though of returning an object with 2 different field instead of making two separated methods, but I couldn't figure out a easy way of doing it.